sections/reading_rust.tex

```diff
@@ -285,16 +294,18 @@ \section{Reading Rust}
 \begin{frame}[fragile]{Trait Based Generics}
 
 \begin{minted}{rust}
-pub fn print_area<Shape: HasArea>(shape: &Shape) {
+pub fn print_area<S: HasArea>(shape: &S) {
```

Comment:
Why `S` and not `Shape`?

Reply:
It's not idiomatic to use a non-fully-uppercase word as type parameter. and `SHAPE: HasArea` looks crappy.